packages/@react-aria/focus/src/useFocusRing.ts

```diff
@@ -50,8 +54,15 @@ export function useFocusRing(props: FocusRingProps = {}): FocusRingAria {
   let updateState = useCallback(() => setFocusVisible(state.current.isFocused && state.current.isFocusVisible), []);
 
   let onFocusChange = useCallback(isFocused => {
-    state.current.isFocused = isFocused;
-    setFocused(isFocused);
+    // ListView needs to not focus when getting focus during scroll and
+    // useFocusVisibleListener() is only aware of modality changes of down events.
+    if (isListView) {
```

Comment:
I need feedback on this change. useFocusVisibleListener() doesn't pick up scroll events. Therefore, FocusRing adds FocusRing classes to Listview when a user has keyboard selected on a ListViewItem, they use the mouse to scroll the listview, and focus has shifted to the ListView because the ListViewItem is out of the dom via the virtualizer. The following code says that if the FocusRing is for a ListView, check to see what the modality is and include that in the logic for saying an item is focused or not. It doesn't set focus if modaility is mouse and the item is focused. It seems to work and pass all the tests, but it seems like the foundation it is built on could be a little smarter to handle this better. I've also noticed that always looking for "scroll" is a lot of events and the code isn't really written to deal with scroll and down events as peers. Suggestions?

P.S. There is a weird interaction still.
Give focus to "Focus Before"
Use the mouse to scroll the ListView so the first item is out of view.
Press tab.
While the ListView is scrolling the first ListViewItem into view, at the same time use the mouse to scroll to the bottom of the list.
The ListView has the focus classes applied to it.
The logic to kept the ListView from getting focus isn't working for this case.

Reply:
Rename this since it would be general to Virtualizer.